desc/magnetic_fields/_current_potential.py

```diff
@@ -842,3 +1042,790 @@ def nfp_loop(j, f):
+        associated data.
+    source_grid : Grid, optional
+        Source grid upon which to evaluate the surface current when calculating
+        the normal field on the plasma surface. Also used to evaluate the
+        virtual casing current, if the plasma has finite plasma currents.
+        Defaults to
+        LinearGrid(M=max(3 * current_potential_field.M_Phi, 30),
+        N=max(3 * current_potential_field.N_Phi, 30), NFP=eq.NFP)
+    eval_grid : _type_, optional
+        Grid upon which to evaluate the normal field on the plasma surface, and
+        at which the normal field is minimized.
+        Defaults to
+        `LinearGrid(M= 30, N= 30, NFP=eq.NFP)`
+    current_helicity : tuple of size 2, optional
+        Tuple of (q,p) used to determine coil topology, where q is the
+        number of poloidal transits a coil makes in one field period and
+        p is the number of toroidal transits a coil makes in one field period.
+        if p is zero and q nonzero, it corresponds to modular coil topology.
+        If both p,q are nonzero, it corresponds to helical coils.
+        If p,q are both zero, it corresponds to windowpane coils.
+        The net toroidal current (when q is nonzero) is set as
+        I = p(G-G_ext)/q/NFP
+    external_field: _MagneticField,
+        DESC `_MagneticField` object giving the magnetic field
+        provided by any coils/fields external to the winding surface.
+        e.g. can provide a TF coilset to calculate the surface current
+        which is needed to minimize Bn given this external coilset providing
+        the bulk of the required net toroidal magnetic flux, by default None
+    external_field_grid : Grid, optional
+        Source grid with which to evaluate the external field when calculating
+        its contribution to the normal field on the plasma surface (if it is a type
+        that requires a source, like a `CoilSet` or a `CurrentPotentialField`).
+        By default None, which will use the default grid for the given
+        external field type.
+    verbose : int, optional
+        level of verbosity, if 0 will print nothing.
+        1 will display Bn max,min,average and chi^2 values for each
+        lambda_regularization.
+        2 will display jacobian timing info
+    normalize : bool, optional
+        whether or not to normalize Bn when printing the Bnormal errors. If true,
+        will normalize by the average equilibrium field strength on the surface.
+    vacuum : bool, optional
+        if True, will not include the contribution to the normal field from the
+        plasma currents.
+    regularization_type : {"simple","regcoil"}
+        whether to use a simple regularization based off of just the single-valued
+        part of Phi, or to use the full REGCOIL regularization penalizing | K | ^ 2.
+        Default is simple, which is much cheaper than the REGCOIL regularization.
+
+    Returns
```

Comment:
This was marked as resolved, but I think this still always returns a list even if it is length 1? I agree with Rory's comment that this would be useful

Reply:
Re-read Rory'c comment, the point was to always have a list output, with length 1 if lambda is a scalar, which is how it is implemented